src/backend/distributed/commands/utility_hook.c

```diff
@@ -1640,3 +1663,46 @@ RunPostprocessMainDBCommand(Node *parsetree)
 		RunCitusMainDBQuery(mainDBQuery->data);
 	}
 }
+
+
+/*
+ * IsStatementSupportedIn2Pc returns true if the statement is supported in 2pc
+ */
+static bool
+IsStatementSupportedIn2Pc(Node *parsetree)
+{
+	NodeTag type = nodeTag(parsetree);
+
+	for (int i = 0; i < sizeof(twoPcSupportedStatements) /
+		 sizeof(twoPcSupportedStatements[0]); i++)
+	{
+		if (type == twoPcSupportedStatements[i].statementType)
+		{
+			return true;
+		}
+	}
+
+	return false;
+}
+
+
+/*
+ * IsStatementMarkDistributedFor2PC returns true if the statement should be marked
+ * as distributed in 2pc
+ */
+static bool
+IsStatementMarkDistributedFor2PC(Node *parsetree)
```

Comment:
Can you rename this to something like `DoesStatementRequireMarkDistributedFor2PC`